database moved to the main spec

SergeyKanzhelev opened the pull request:
@danielkhan I suggest something like this - move one protocol at a time to the main spec. So every protocol can be reviewed separately.

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,47 @@
+Note, Redis, Cassandra, HBase and other storage systems may reuse the same
+attribute names.
+
+| Attribute name | Notes and examples |
```
Are the naming conventions for attributes already agreed upon? (like db.type, etc.)

flands replied:
They are taken directly from here: https://github.com/opentracing/specification/blob/master/semantic_conventions.md. Unless there is a good reason, should probably stick with them.

flands commented on semantic-conventions.md:
```diff
@@ -1 +1,47 @@
+Note, Redis, Cassandra, HBase and other storage systems may reuse the same
+attribute names.
+
+| Attribute name | Notes and examples |
```
Should we have a notion of required versus not? Something like `db.type` required, rest optional?

danielkhan replied:
Yes - in my opinion, there should definitely be a distinction between mandatory and optional.
Otherwise, the whole concept of typed spans would not play out well.
I am still fond of merging the PR regardless and filling the blanks later because each attribute can be a discussion on its own.

There is already existing work on mandatory vs. required attributes that could be incorporated then https://docs.google.com/spreadsheets/u/1/d/1H0S0BROOgX7zndWF_WL8jb9IW1PN7j3IeryekhX5sKU/edit#gid=0

flands commented on semantic-conventions.md:
```diff
@@ -1 +1,47 @@
+
+## Databases client calls
+
+Calls to databases should be tracked as `Client` spans.
```
Terms used must be clearly defined and used consistently. For database client call the SpanKind MUST be client.

flands commented on semantic-conventions.md:
```diff
@@ -1 +1,47 @@
+represents calls that use well-known protocols like HTTP or database calls.
+Depending on the protocol and the type of an operation, additional information
+is needed to represent and analyze a span correctly in monitoring systems. It is
+also important to unify how this attribution is made in different languages. So
```
s/So/This way the/

flands commented on semantic-conventions.md:
```diff
@@ -1 +1,47 @@
+| `db.statement` | A database statement for the given database type. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. |
+| `db.user`      | Username for accessing database. E.g., `"readonly_user"` or `"reporting_user"` |
+
+For database client calls, peer information SHOULD be collected.
```
Why is this a separate section? I assume most of these are SHOULD. Also, we should capitalize terms or not for consistency

SergeyKanzhelev replied:
`peer.*` attributes are common. I rephrased

tylerbenson commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+| `component`    | Database driver name or database name (when known) `JDBI`, `jdbc`, `odbc`, `postgreSQL`. | Yes       |
+| `db.type`      | Database type. For any SQL database, `"sql"`. For others, the lower-case database category, e.g. `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
+| `db.instance`  | Database instance name. E.g., In java, if the jdbc.url=`"jdbc:mysql://127.0.0.1:3306/customers"`, the instance name is `"customers"`. | Yes       |
+| `db.statement` | A database statement for the given database type. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. | Yes       |
```
Might be worth mentioning that this could/should be a sanitized representation and not the raw query.

tylerbenson commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+| Attribute name  | Notes and examples                                           | Required |
+| :-------------- | :----------------------------------------------------------- | -------- |
+| `peer.address`  | JDBC substring like `"mysql://prod-db:3306"`                 | Yes      |
+| `peer.hostname` | Remote hostname. `localhost`                                 | Yes      |
```
In both of these, perhaps use something more obviously a domain name (`db.example.com`) and consistent between the two.

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+In OpenTelemetry spans can be created freely and it’s up to the implementor to
+annotate them with attributes specific to the represented operation. Spans
+represent specific operations in and between systems. Some of these operations
+represents calls that use well-known protocols like HTTP or database calls.
```
```suggestion
represent calls that use well-known protocols like HTTP or database calls.
```

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+annotate them with attributes specific to the represented operation. Spans
+represent specific operations in and between systems. Some of these operations
+represents calls that use well-known protocols like HTTP or database calls.
+Depending on the protocol and the type of an operation, additional information
```
```suggestion
Depending on the protocol and the type of operation, additional information
```

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+Depending on the protocol and the type of an operation, additional information
+is needed to represent and analyze a span correctly in monitoring systems. It is
+also important to unify how this attribution is made in different languages.
+This way the operator will not need to learn specifics of a language and
```
```suggestion
This way, the operator will not need to learn specifics of a language and
```

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+For database client call the `SpanKind` MUST be `Client`.
+
+Span `name` should be set to low cardinality value representing the statement
+executed on database. It may be stored procedure name (without argument), sql
```
```suggestion
executed on the database. It may be stored procedure name (without argument), sql
```

danielkhan commented on semantic-conventions.md:
```diff
@@ -1 +1,49 @@
+| `db.user`      | Username for accessing database. E.g., `"readonly_user"` or `"reporting_user"` | No        |
+
+For database client calls, peer information can be populated and interpreted as
+following:
```
```suggestion
follows:
```

danielkhan commented on work_in_progress/typedspans/README.md:
```diff
@@ -69,20 +69,17 @@ Represents an inbound HTTP request.
-- Endpoint
-- Statement
+
+These attributes are not covered in [main spec](../semantic-conventions.md):
```
```suggestion
These attributes are not covered in the [main spec](../semantic-conventions.md):
```

SergeyKanzhelev commented:
CC: @open-telemetry/specs-approvers

SergeyKanzhelev commented:
@flands I think I addressed all of your feedback. I'll merge to the next portion of attributes can be moved over. Don't hesitate to file issues if anything else feels wrong.
